.github/workflows/remove-wheels.yml

```diff
@@ -0,0 +1,78 @@
+name: Remove old wheels
+
+on:
+  # Run daily at 1:23 UTC
+  schedule:
+  - cron:  '23 1 * * *'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+# Needed for micromamba pickup
+defaults:
+  run:
+    shell: bash -l {0}
+
+env:
+  N_LATEST_UPLOADS: 5
+  ANACONDA_USER: "scientific-python-nightly-wheels"
+
+jobs:
+  remove:
+
+    runs-on: ubuntu-latest
+    environment:
+      name: remove-old-wheels
+
+    steps:
+      - name: Install micromamba and anaconda-client
+        uses: mamba-org/setup-micromamba@v1
+        with:
+          environment-name: remove-wheels
+          create-args: >-
+            python=3.11
+            anaconda-client=1.11.2
+
+      - name: Show environment
+        run: env
+
+      - name: Show CLI API info
+        run: |
+          anaconda show --help
+          echo ""
+          anaconda remove --help
+
+      - name: Query package index for packages
+        run: |
+          anaconda show "${ANACONDA_USER}" &> >(grep "${ANACONDA_USER}/") | \
+              awk '{print $1}' | \
+              sed 's|.*/||g' > package-names.txt
+
+      - name: Remove old uploads to save space
+        run: |
+          # Remove all _but_ the last ${N_LATEST_UPLOADS} package versions
+          # N.B.: `anaconda show` places the newest packages at the bottom of the output
+          # of the 'Versions' section and package versions are preceded with a '   + '.
+
+          if [ -s package-names.txt ]; then
+              while LANG=C IFS= read -r package_name ; do
```

Comment:
Alternative spelling that may be a bit easier on the eye:

```
for pkg in $(cat package-names.txt); do
  ... something with $pkg ...
done
```

Reply:
I'll fully admit that my use of the `while IFS= read -r line` pattern comes from blindly following [Shellcheck SC2013](https://www.shellcheck.net/wiki/SC2013) from years ago and at this point it is drilled into the fingers. I don't mind shifting things if it will help with the maintainability, but I think a nicer goal would be to eventually learn enough of the underlying `binstar_client` Python API from `anaconda-client` to rewrite things in not-a-shell language (assuming that somehow isn't worse for readability). :)

Reply:
I agree with the comment above, it's so much easier to read the none "while" version.